cogs/hideout/addbot.py

```diff
@@ -116,19 +116,28 @@ async def on_member_remove(self, member: discord.Member):
 
         queue_channel: discord.TextChannel = member.guild.get_channel(QUEUE_CHANNEL)  # type: ignore
 
+        _bot_ids = await self.bot.pool.fetch('SELECT bot_id FROM addbot WHERE owner_id = $1 AND added = TRUE', member.id)
+        bots = [_ent for _ent in map(lambda ent: member.guild.get_member(ent['bot_id']), _bot_ids) if _ent is not None]
+
```

Comment:
This should be after line 123, probably inside `if mem:`. There's no escaping duplicate code. Also, `member` in this logic path is the bot, not the owner, so this would not work regardless. You must use ``mem`` instead.

```py
has_bots = await self.bot.pool.fetchval("SELECT EXISTS(SELECT 1 FROM addbot WHERE owner_id = $1 AND added = TRUE)", mem.id)

if not has_bots:
```